### ISSUE TITLE
BelongsToMany options not working

### ISSUE THREAD

**Issue by DanielLavoie90** [optimistdigital/nova-multiselect-field#68]
I'm on Nova 3.12.1 and I need to overwrite the fillUsing you have on the belongsToMany function cause the callback method you use is returning a callable when it should not. 

When I replace fillUsing like this (remove the return function) it all work fine:
```php
->fillUsing(function ($request, $model, $requestAttribute, $attribute) {
//return function () use ($request, $model, $attribute) {     <-- removing this
    // Validate
    if (!is_callable([$model, $attribute])) {
        throw new RuntimeException("{$model}::{$attribute} must be a relation method.");
    }

    $relation = $model->{$attribute}();

    if (!method_exists($relation, 'sync')) {
        throw new RuntimeException("{$model}::{$attribute} does not appear to model a BelongsToMany or MorphsToMany.");
    }

    // Sync
    $relation->sync($request->get($attribute) ?? []);
//};  <-- And removing this (of course)
})
```

**Comment by DanielLavoie90** [comment]
Since removing the return of a function only works when updating, I switched the return for :
`$model::saved(function($model){`
With this, all works fine in all cases ;)

**Comment by Tarpsvo** [comment]
Hi @DanielLavoie90!

Thanks a lot for the feedback, the function return was indeed off. I implemented the changes you suggested and released a new version, 1.10.2.

Thanks and good luck!
